### DISCUSSION TITLE
Adding a bit of text to the footer

### DISCUSSION THREAD

**Discussion by PlasmaHH** [Textualize/textual#5286] [Q&A] [answered]
I have been looking for a way to add a bit of text to the footer (more specifically replacing the command palette key description in the right corner) but found no obvious easy way (apologies if I missed it, there is just so much cool features).

I came up with the following hack (transcribed from another computer, might contain typOs):

```python
class StatusFooter( Footer ):
    def __init__( self ):
        super().__init__()
        self.status = Label("FOO")
        self.show_command_palette = False
        self.status.styles.dock = "right"

    def compose( self ):
        yield from super().compose()
        yield self.status
```

From looking at _footer.py what I thought would happen is that all the yields there assemble the normal footer as usual and just add mine later. It also first looks like it works fine, however if you remove the dock = right part the label is positioned to the very left of the footer overwriting everything else there.

How can I have an extra bit of text (status messages etc.) as the "next" element in the footer after the keybinding descriptions?

**Comment by TomJGooding** [comment]
The `Footer` is styled as a grid, where [the number of columns is set to the number of bindings](https://github.com/Textualize/textual/blob/3a8eeddbef8d2d81ccbe709fe97724422d14c7d8/src/textual/widgets/_footer.py#L220). Without the `dock` styling on your status label (the same as the command palette key you're replacing), it will be added to the next row in the grid.

**Reply by PlasmaHH** [reply]
Ok, so increasing self.styles.grid_size_columns by 1 before I `yield self.status` should do the trick, not sure if this is the "canonical" way to do it though...

**Reply by TomJGooding** [reply, marked as answer]
Sorry I think I misunderstood where you wanted the label to be positioned. I thought having the status docked to the right would make more sense than displaying it right beside the bindings!

I think incrementing the `grid_size_columns` is probably your best solution in that case, given the styling constraints of the grid. There's not a "canonical" way of extending the footer AFAIK.

**Reply by PlasmaHH** [reply]
The dock thing was to mainly to illustrate my point of things being overwritten rather than layed out together... but thinking about it, it may indeed make more sense to dock it to the right per default...
